Source/Core/VideoCommon/BoundingBox.h

```diff
@@ -4,40 +4,44 @@
 
 #pragma once
 
+#include <array>
+#include <vector>
+
 #include "Common/CommonTypes.h"
 
 class PointerWrap;
 
-// Bounding Box manager
-namespace BoundingBox
-{
-// Indicates a coordinate of the bounding box.
-enum class Coordinate
+using BBoxType = s32;
+static constexpr u32 NUM_BBOX_VALUES = 4;
+
+class BoundingBox
 {
-  Left,    // The X coordinate of the left side of the bounding box.
-  Right,   // The X coordinate of the right side of the bounding box.
-  Top,     // The Y coordinate of the top of the bounding box.
-  Bottom,  // The Y coordinate of the bottom of the bounding box.
-};
+public:
+  virtual ~BoundingBox() = default;
 
-// Enables bounding box.
-void Enable();
+  bool IsEnabled() { return m_is_active; }
+  void Enable();
+  void Disable();
 
-// Disables bounding box.
-void Disable();
+  void Flush();
 
-// Determines if bounding box is enabled.
-bool IsEnabled();
+  u16 Get(int index);
+  void Set(int index, u16 value);
 
-// Gets a particular coordinate for the bounding box.
-u16 GetCoordinate(Coordinate coordinate);
+  void DoState(PointerWrap& p);
 
-// Sets a particular coordinate for the bounding box.
-void SetCoordinate(Coordinate coordinate, u16 value);
+  virtual bool Initialize() = 0;
 
-// Updates all bounding box coordinates.
-void Update(u16 left, u16 right, u16 top, u16 bottom);
+protected:
+  virtual std::vector<BBoxType> Read(int index, int length) = 0;
+  virtual void Write(int index, const std::vector<BBoxType> values) = 0;
```

Comment:
```suggestion
  virtual void Write(int index, const std::vector<BBoxType>& values) = 0;
```

Perhaps leave a general TODO or Note comment that this vector can likely be turned into a `std::span` whenever we update to C++20.